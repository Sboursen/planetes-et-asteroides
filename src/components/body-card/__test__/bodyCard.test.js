import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import BodyCard from '../bodyCard';

describe('BodyCard', () => {
  const handleBodyCardClick = (id) => id;
  afterEach(cleanup);

  it('renders correctly in the DOM', () => {
    const tree = renderer
      .create(
        <BodyCard
          bodyId="terre"
          name="Earth"
          moonsNumber={1}
          orbitsAround="Sun"
          handleClick={(id) => handleBodyCardClick(id)}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the given name', () => {
    render(
      <BodyCard
        bodyId="terre"
        name="Earth"
        moonsNumber={1}
        orbitsAround="Sun"
        handleClick={(id) => handleBodyCardClick(id)}
      />,
    );

    const nameElement = screen.getByText('Earth');
    const { classList } = nameElement;

    expect(classList).toContain('text-2xl');
  });

  it('renders with the given moonsNumber', () => {
    render(
      <BodyCard
        bodyId="terre"
        name="Earth"
        moonsNumber={1}
        orbitsAround="Sun"
        handleClick={(id) => handleBodyCardClick(id)}
      />,
    );

    const nameElement = screen.getByText(/moons$/);
    const { classList } = nameElement;

    expect(classList).toContain('whitespace-nowrap');
  });
});
