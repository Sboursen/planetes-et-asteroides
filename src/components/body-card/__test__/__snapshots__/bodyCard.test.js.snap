// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyCard renders correctly in the DOM 1`] = `
<div
  className="relative flex flex-col gap-3 p-5 w-full"
>
  <button
    className="absolute top-2 right-2 hover:text-header hover:bg-white rounded-full"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <h2
    className="text-2xl font-medium w-full text-center"
  >
    Earth
  </h2>
  <div
    className="w-full flex flex-row justify-around items-center"
  >
    <span
      className="whitespace-nowrap"
    >
      1 moons
    </span>
    <div
      className="group hover:animate-spin hover:cursor-default whitespace-nowrap rounded px-2"
    >
      <div
        className="relative w-20 h-20 rounded-full border-white border-2 flex justify-center items-center"
      >
        <span
          className="flex justify-center items-baseline pt-1"
        >
          SUN
        </span>
        <span
          className="absolute group-hover:animate-ping-fast top-[7px] right-[7px] p-1 rounded-full bg-white hover:bg-red-400"
        />
      </div>
    </div>
  </div>
</div>
`;
